src/microsoft-extensions-ai/abstraction-implementations/AbstractionImplementationExamples/Middleware.cs

```diff
@@ -0,0 +1,54 @@
+using System.ComponentModel;
+using System.Diagnostics;
+using Microsoft.Extensions.AI;
+using Microsoft.Extensions.Caching.Distributed;
+using OpenTelemetry.Trace;
+
+public partial class AbstractionSamples
+{
+    public static async Task Middleware() 
+    {
+        // Configure OpenTelemetry Exporter
+        var sourceName  = Guid.NewGuid().ToString();
+        var activities = new List<Activity>();
+
+        var tracerProvider = OpenTelemetry.Sdk.CreateTracerProviderBuilder()
+            .AddSource(sourceName)
+            .AddInMemoryExporter(activities)
+            .Build();
+
+        // Configure cache
+        IDistributedCache cache = new InMemoryCacheStorage();
```

Comment:
This can use:
```
new MemoryDistributedCache(Options.Create(new MemoryDistributedCacheOptions()))
```
instead of needing a custom `InMemoryCacheStorage`.